### PR TITLE
features: logging in and out; viewing, filtering, searching artwork. 

### PR DESCRIPTION
**Pull request by z-rodney**
first dev to main PR including work from week 1: backend routes; db set up; viewing, filtering and searching art; logging in and out
Version 1.0

### CODE REVIEW

File: server/db/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+User.hasOne(Cart)
+
+Artwork.belongsToMany(Cart, {through: 'cartItem'})
+
+const syncDB = async(forceSeed = false)=> {
+  // Took out force: true so the db didn't drop the seeded data every time
+  await db.sync({ force: forceSeed });
+};
+
+module.exports = {
```
Comment (peetklecha) [inline server/db/index.js:57]: instead of re-listing all the models here, you could just spread require("./models")

File: server/api/genres.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const express = require('express')
+const router = express.Router()
+const { Genre, Artwork } = require('../db')
+
```
Comment (peetklecha) [inline server/api/genres.js:4]: i'm a little surprised to see that your "get all genres" route includes all the products with each genre (so if this site were to scale up, a massive amount of data) but then your get-single-genre does *not* include the associated products. sort of the opposite of what you'd normally expect.

**Review by leafoflegend — APPROVED** [review]
Models look good! I think some linting standards need to be sorted out.

Nothing sticks out to me as critically wrong. The seed file is certainly brittle, and theres some definite overuse and misunderstanding of `async`.

Keep it up!

File: client/components/AllArtwork.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    this.setState({
+      artworks: this.props.artworks,
+      artists: this.props.artists,
+      genres: this.props.genres
+    });
+  }
+
+  async changeFilter(ev) {
+    const value = ev.target.value === 'DEFAULT' ? '' : ev.target.value;
+    await this.setState({
```
[the remark this answers is not in the document]
Comment (peetklecha) [reply client/components/AllArtwork.js:43]: sort of beside the point, but my understanding is that `setState` *is* asynchronous, but unusually for an asynchronous function, it doesn't return a `Promise` (or take a callback).

[the remark this answers is not in the document]
Comment (leafoflegend) [reply client/components/AllArtwork.js:43]: Very true - thats getting into some implementation details that aren't our business. As a user of React, I believe you are supposed to treat it as synchronous. Either way - this is no place for `async` or `await`

File: server/db/index.js (hunk 1 above)
Comment (zoegreene) [reply server/db/index.js:57]: Not sure I understand. What is the syntax of that?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
+node_modules
+server/public/bundle.js
```
[the remark this answers is not in the document]
Comment (zoegreene) [reply .gitignore:2]: What do you mean by that?

File: seed.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+  [22,[15]],
+  [23,[4,15]],
+  [24,[16,17]],
+  [25,[5]],
+  [26,[19]],
+  [27,[5]],
+  [28,[18]],
+  [29,[20]],
+  [30,[15]]
+]
```
[the remark this answers is not in the document]
Comment (zoegreene) [reply seed.js:184]: Do you have any suggestions for how to clean it up? It's a lot of data to add!

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (leafoflegend) [reply .gitignore:2]: Globs are

```
server/public/*.js
```

File: seed.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (leafoflegend) [reply seed.js:184]: One thought is to use `faker` and dynamically generate data. Another is to just have a lot of these magic numbers stored in variables. The fact that you are manually writing these numbers and have to track them is really what makes this insane.

**Pull request merged**